src/tasks/FileUploadUtil/Interfaces/IProgress.ts

```diff
@@ -0,0 +1,26 @@
+/* eslint-disable @typescript-eslint/type-annotation-spacing */
+import { GraphClientError } from "../../../GraphClientError";
+import { Range } from "../../../Range";
+import { UploadResult } from "../UploadResult";
+
+/**
+ * Interface enabling progress handling with callbacks.
+ */
+export interface Progress {
+	/**
+	 * Parameters that are passed into the progress, completed, failure callback options.
+	 */
+	extraCallbackParams?: unknown;
```

Comment:
shouldn't this be an array of unknown?

Reply:
unknown can be anything. Changed the name to `extraCallbackParam`. The extra param could be anything such as an object/type specific to the user application or an array of unknown.

Reply:
thanks! yeah the plural name was leading me to think it should be an array. If a parameter/property needs to be an array of something, but we don't know what, it's always good to have an array of unknown as the type instead of just unknown. It tells the consumer we're at least expecting an array and we get transpile time validation for the array prototype (the "first level of the object model"). Better than nothing :)

Reply:
Thank you for the reasoning and I agree that if the `extraCallbackParam` was only expecting an array, then it should be defined explicitly.